pkg/core/blockchain.go

```diff
@@ -401,71 +401,45 @@ func (bc *Blockchain) persistBlock(block *Block) error {
 	if err := assets.commit(batch); err != nil {
 		return err
 	}
-	if err := bc.PutBatch(batch); err != nil {
+	if err := bc.memStore.PutBatch(batch); err != nil {
 		return err
 	}
 
-	bc.persistedHeight = block.Index
+	atomic.StoreUint32(&bc.blockHeight, block.Index)
```

Comment:
nit:
in godoc found for `&bc.blockHeight`
"Write access should only happen in AddBlock()." Probably it's not valid anymore.